libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Generators;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    /// <summary>
+    /// An <see cref="IBot"/> implementation that hosts an <see cref="AdaptiveDialog"/>.
+    /// </summary>
+    public class AdaptiveDialogBot : IBot
+    {
+        private const string DefaultLocale = "en-US";
+        private const string DefaultLg = "main.lg";
+        private const string DefaultDialogName = "main.dialog";
+
+        private static readonly ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager> _languageGeneratorManagers = new ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager>();
+
+        private readonly string _defaultLocale;
+        private readonly string _rootDialogId;
+
+        private readonly ILogger _logger;
+        private readonly ResourceExplorer _resourceExplorer;
+        private readonly BotFrameworkAuthentication _botFrameworkAuthentication;
+
+        private readonly SkillConversationIdFactoryBase _skillConversationIdFactoryBase;
+        private readonly ConversationState _conversationState;
+        private readonly UserState _userState;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AdaptiveDialogBot"/> class.
+        /// </summary>
+        /// <param name="resourceExplorer">The Bot Builder <see cref="ResourceExplorer"/> to load the <see cref="Dialog"/> from.</param>
+        /// <param name="rootDialogId">The id of the dialog to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="defaultLocale">The default locale for this bot.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        /// <param name="storage">The <see cref="IStorage"/> implementation to use for this <see cref="Dialog"/>.</param>
```

Comment:
Add default to documentation. 
Also, I don't understand the <see cref="Dialog"/> reference, did you mean another class? (I don't see any references to storage in Dialog)